### PR TITLE
Render rasters over time.

### PR DESCRIPTION
**Pull request by DamianRodziewicz**

### CODE REVIEW

**Comment by DamianRodziewicz** [comment]
@mateuszwyszynski please go through first round of review. I need to update documentation and rewrite timeSinceFire later but for now I'd like you to take a first look at the code.

**Comment by mateuszwyszynski** [comment]
We need to add `gdalUtils` package to the _Imports_ section in DESCRIPTION file.

File: R/rastersOverTime.R [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    rasterIndex <- if (is.null(rasterIndexValue())) {
+      1
+    } else {
+      rasterIndexValue() / rasterStepSize + 1
+    }
+
+    raster <- rasters[[rasterIndex]]
+
+    Cache(gdal2Tiles, raster, outputPath = file.path("www", session$ns("map-tiles")),
+          zoomRange = 1:10, colorTextFile = asPath(colorTableFile),
```
Comment (mateuszwyszynski) [inline R/rastersOverTime.R]: `colorTextFile` should become `colorTableFile`

**Review by mateuszwyszynski — APPROVED** [review]

**Comment by DamianRodziewicz** [comment]
This PR will be merged only after all style issues are fixed and timeSinceFire module is changed to use rastersOverTime module.

**Comment by achubaty** [comment]
Travis is failing for osx because of a problem with the `rJava` package, which is a dependency of `xlsx`.

I had encountered similar errors locally but running `sudo R CMD javareconf` and doing `Sys.setenv(LD_LIBRARY_PATH = ":@JAVA_LD@")` seemed to do the trick. R CMD check now passes on macOS locally, but with the following note:

```r
checking R code for possible problems ... NOTE
initialMap: no visible binding for global variable ‘demoArea’
largePatches : <anonymous>: no visible binding for global variable ‘.N’
timeSinceFire: no visible binding for global variable
  ‘shpStudyRegionFull’
timeSinceFire: no visible binding for global variable ‘providers’
timeSinceFire: no visible binding for global variable
  ‘timeSinceFirePalette’
timeSinceFire: no visible binding for global variable ‘maxAge’
timeSinceFire: no visible binding for global variable ‘colorTableFile’
Undefined global functions or variables:
  .N colorTableFile demoArea maxAge providers shpStudyRegionFull
  timeSinceFirePalette
R CMD check results
0 errors | 0 warnings | 1 note 
```

This is likely a transient error on travis, which will hopefully be fixed soon.

**Comment by achubaty** [comment]
**RE: issues on appveyor**

These seem to be related to the `XML` package and may be transient, but I can't reproduce right now because I don't have access to a Windows machine.

```r
* installing *source* package 'SpaDES.shiny' ...
** R
** inst
** byte-compile and prepare package for lazy loading
Warning in gzfile(file, "rb") :
  cannot open compressed file 'C:/Users/appveyor/AppData/Local/Temp/1/RtmpygeTJS/RLIBS_86473befcf/XML/R/XML.rdx', probable reason 'No such file or directory'
Error in gzfile(file, "rb") : cannot open the connection
Error : unable to load R code in package 'XML'
ERROR: lazy loading failed for package 'SpaDES.shiny'
* removing 'C:/projects/spades-shiny/SpaDES.shiny.Rcheck/SpaDES.shiny'
```

@DamianRodziewicz can you please run R CMD check locally on Windows to see that whether it passes?

**Comment by DamianRodziewicz** [comment]
@achubaty - R CMD check passes on Windows machine for me locally. I'll fix the notes

**Comment by achubaty** [comment]
Checks on Appveyor are now passing for r-release, but r-devel still fails for unknown reason.
Checks on Travis for macOS are <s>in queue...</s> passing.

**Comment by achubaty** [comment]
appveyor r-devel is likely to be failing spuriously at this point, but all other tests pass.

**Pull request merged**